Complete Product page

blobinaticuber opened the pull request:
Add link on product name to go to product page
Add product page for each product where you can see name, image, price, ID
Add links to product page to check out or continue shopping

blobinaticuber committed:
clicking product name brings you to product page

blobinaticuber committed:
add table for other attributtes

blobinaticuber committed:
add links to product.jsp

blobinaticuber committed:
style buttons

iansteyn reviewed (changes requested):
Add to cart link doesn't work properly.

blobinaticuber committed:
still not working, but better code

blobinaticuber committed:
minor refactoring

iansteyn committed:
improved indentation

iansteyn committed:
bugfix in listprod - change %s to %d

productId was being incorrectly passed to the product.jsp page as a string instead of an int

iansteyn committed:
change id to productId in productLink

better consistency that way

iansteyn committed:
Reduce clutter, improve commenting and variable naming

Replaced connection code with getConnection() function from jdbc.jsp, improved overall code organization

iansteyn committed:
Big html formatting improvements

I just took all the html out of out.printlns and string.formats, and made it normal html with <%= expressions %> inserted. A lot more readable. This is new to me too.

iansteyn committed:
changed productId back to id for ACTUAL consistency

iansteyn committed:
Various jsp compiling bugfixes

several bugs related to variable names etc happened because of my refactoring. Silly me

iansteyn committed:
Add ability to clear cart

Add clearcart.jsp and appropriate link on showcart.jsp

iansteyn committed:
wrapped buttons in anchor(link) tags instead of form tags

now the parameters will actually go through

iansteyn reviewed (changes requested):
## ✅ What works now
- The add-to-cart button actually works now. The solution was to wrap it in `<a href="..."></a>` anchor tags instead of form tags.
- Also I made some formatting/indentation/general organization changes to make the code easier to work on.

## ⚠️ What still needs to be done
- `product` objects in the database can have actual image files stored in them in the attribute `productImage` (specifically, in our case, chai has both an imageURL *and* such an image - see line 243 of the ddl)
- Part of the instructions for this file are:
- > **+3 marks** - for displaying an image from the binary field `productImage` by providing an `img` tag and modifying the `displayImage.jsp/php` file.
- so please do this part

blobinaticuber committed:
add product description to product page

blobinaticuber committed:
change header name

blobinaticuber committed:
add comment

iansteyn committed:
change name in title tags

iansteyn committed:
Add img tag which uses displayImage as source

iansteyn committed:
bugfix: removed an extra c from displayImage.jsp

iansteyn committed:
Add link to shop.html

iansteyn reviewed (approved):
Second image should work now...
